test/suite/test_local01.py

```diff
@@ -0,0 +1,249 @@
+
+        # blocks filled with 'a', etc.
+        a_block = ('a' * block_size).encode()
+        b_block = ('b' * block_size).encode()
+        c_block = ('c' * block_size).encode()
+        file_size = nblocks * block_size
+
+        # write all blocks as 'a', but in reverse order
+        for pos in range(file_size - block_size, 0, -block_size):
+            fh.fh_write(session, pos, a_block)
+
+        # write the even blocks as 'b', forwards
+        for pos in range(0, file_size, block_size * 2):
+            fh.fh_write(session, pos, b_block)
+
+        # write every third block as 'c', backwards
+        for pos in range(file_size - block_size, 0, -block_size * 3):
+            fh.fh_write(session, pos, c_block)
+        fh.close(session)
+
+        in_block = bytes(block_size)
+        fh = local.ss_open_object(session, location, 'abc', StorageSource.open_readonly)
+
+        # Do some spot checks, reading non-sequentially
+        fh.fh_read(session, 500 * block_size, in_block)  # divisible by 2, not 3
+        self.assertEquals(in_block, b_block)
+        fh.fh_read(session, 333 * block_size, in_block)  # divisible by 3, not 2
+        self.assertEquals(in_block, c_block)
+        fh.fh_read(session, 401 * block_size, in_block)  # not divisible by 2 or 3
+        self.assertEquals(in_block, a_block)
+
+        # Read the whole file, backwards checking to make sure
+        # each block was written correctly.
+        for block_num in range(nblocks - 1, 0, -1):
+            pos = block_num * block_size
+            fh.fh_read(session, pos, in_block)
+            if block_num % 3 == 0:
+                self.assertEquals(in_block, c_block)
+            elif block_num % 2 == 0:
+                self.assertEquals(in_block, b_block)
+            else:
+                self.assertEquals(in_block, a_block)
+        fh.close(session)
+
+    def create_in_loc(self, loc, objname):
+        session = self.session
+        fh = self.local.ss_open_object(session, loc, objname, StorageSource.open_create)
+        fh.fh_write(session, 0, 'some stuff'.encode())
+        fh.close(session)
+
+    def check(self, loc, prefix, limit, expect):
+        # We don't require any sorted output for location lists,
+        # so we'll sort before comparing.'
+        got = sorted(self.local.ss_location_list(self.session, loc, prefix, limit))
+        expect = sorted(expect)
+        self.assertEquals(got, expect)
+
+    def test_local_locations(self):
+        # Test using various buckets, clusters
+
+        session = self.session
+        local = self.conn.get_storage_source('local_store')
+        self.local = local
+        os.mkdir("objects1")
+        os.mkdir("objects2")
+
+        # Any of the activity that happens in the various locations
+        # should be independent.
+        location1 = local.ss_location_handle(session,
+            'cluster="cluster1",bucket="./objects1",kmsid="k1"')
```

Comment:
This should be part of `wiredtiger_open`. That is exactly the set of configs I'm adding in 7235. I think this should go away at that point?

Reply:
I don't think it goes away.  The storage source needs to be told at some point about the bucket, kmsid etc., and this is that point.  However, these calls would normally be taking place in the WT library.  Normally, an application would not be talking to the storage source API directly, this test does that just to test the implementation.
When we have enough infrastructure in place, we can put these bucket, kmsid parameters into wiredtiger open and we can use the normal WT API, and never call any of these directly.